pydasi/dasi/dasi.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 European Centre for Medium-Range Weather Forecasts (ECMWF)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import fsencode
+
+from ._dasi_cffi import DASIException, ffi, lib
+from .key import Key
+from .query import Query
+from .retrieve import Retrieve
+from .utils import DEBUG, getLogger
+
+logger = getLogger(__name__)
+logger.setLevel(DEBUG)
+
+
+class Dasi:
+    """
+    This is the Dasi::Dasi.
+    """
+
+    def __init__(self, config_path: str):
+        if config_path:
+            logger.info("Config file: %s", config_path)
+            self._new_session(config_path)
+        else:
+            err_msg = "Could not find the config file:\n{}".format(config_path)
+            raise DASIException(err_msg)
+
+    @property
```

Comment:
What is this used for? It is a surprisingly complicated function for something that isn't in a generic base class...